pos-print/src/main/java/com/clouway/pos/print/adapter/http/PrintServiceV2.kt

```diff
@@ -0,0 +1,54 @@
+package com.clouway.pos.print.adapter.http
+
+import com.clouway.pos.print.core.*
+import com.clouway.pos.print.transport.GsonTransport
+import com.google.inject.name.Named
+import com.google.sitebricks.At
+import com.google.sitebricks.headless.Reply
+import com.google.sitebricks.headless.Request
+import com.google.sitebricks.headless.Service
+import com.google.sitebricks.http.Get
+import com.google.sitebricks.http.Post
+import org.slf4j.LoggerFactory
+import javax.inject.Inject
+
+/**
+ * @author Tsvetozar Bonev (tsvetozar.bonev@clouway.com)
+ */
+@Service
+@At("/v2/receipts/req/print")
+class PrintServiceV2 @Inject constructor(private var repository: ReceiptRepository,
+                                         private var queue: PrintQueue) {
+  private val logger = LoggerFactory.getLogger(PrintServiceV2::class.java)
+
+  @Post
+  fun printReceipt(request: Request): Reply<*> {
+    return try {
+      val dto = request.read(ReceiptDTO::class.java).`as`(GsonTransport::class.java)
+      val receiptId = repository.register(dto.receipt)
+
+      queue.queue(PrintReceiptRequest(dto.receipt, dto.sourceIp, dto.fiscal))
+
+      logger.info("Receipt queued for printing")
+      Reply.with(receiptId).ok()
```

Comment:
status 202 is better to be returned as it will indicate that background processing will be performed.

https://httpstatuses.com/202

Reply:
Changed to return status 202.